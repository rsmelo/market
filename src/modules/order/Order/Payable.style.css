.payable {
  padding: 1em;
  border-radius: var(--radius-default);
  background-color: var(--color-10-grey);
  margin: 0.5em 0;
}

.payable:not(:first-child) {
  margin: 0.5em 0 0.5em 0.5em;
}

.value {
  font-weight: lighter;
}

.item {
  line-height: 1.6;
}
