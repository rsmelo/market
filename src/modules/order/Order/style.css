.container {
  display: flex;
  justify-content: center;
  padding: 1em 2em;
}

.title {
  font-size: 22px;
  font-weight: bold;
  line-height: 1.4em;
  color: var(--color-dark-grey);
}

.transaction {
  margin: 1em 0;
}

.payablesTitle {
  color: var(--color-dark-grey);
  margin: 1em 0 0;
  font-size: 16px;
  font-weight: bold;
}

.payablesContainer {
  display: flex;
  flex-flow: wrap;
  width: 100%;
  justify-content: center;
}

.buttonContainer {
  display: flex;
  flex-direction: row-reverse;
  padding: 0.8em 0;
}

.value {
  font-weight: lighter;
}

.item {
  line-height: 1.6;
}
