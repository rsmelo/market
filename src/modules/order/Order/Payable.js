import React from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'

import formatCurrency from '../../../utils/formatCurrency'
import style from './Payable.style.css'

const statuses = {
  paid: 'pago',
  waiting_funds: 'aguardando fundos',
}
const Payable = ({
  status,
  amount,
  fee,
  recipientId,
  paymentDate,
  dateCreated,
}) => (
  <div className={style.payable}>
    <ul>
      <li>
        ID do recebedor: {recipientId}
      </li>
      <li>
        Valor: {formatCurrency(amount)}
      </li>
      <li>
        Taxas: {formatCurrency(fee)}
      </li>
      <li>
        Status: {statuses[status]}
      </li>
      <li>
        Data da criação: {moment(dateCreated).format('DD/MM/YYYY')}
      </li>
      <li>
        Data do pagamento: {moment(paymentDate).format('DD/MM/YYYY')}
      </li>
    </ul>
  </div>
)

Payable.propTypes = {
  status: PropTypes.string.isRequired,
  amount: PropTypes.number.isRequired,
  fee: PropTypes.number.isRequired,
  recipientId: PropTypes.string.isRequired,
  paymentDate: PropTypes.string.isRequired,
  dateCreated: PropTypes.string.isRequired,
}
export default Payable
