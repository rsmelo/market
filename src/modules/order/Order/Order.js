import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

import style from './style.css'
import Payable from './Payable'
import ButtonLink from '../../../components/ButtonLink'
import formatCurrency from '../../../utils/formatCurrency'

const statuses = {
  processing: 'processando',
  authorized: 'autorizado',
  paid: 'pago',
  refunded: 'reembolsado',
  waiting_payment: 'aguardando pagamento',
  pending_refund: 'reembolso pendente',
  refused: 'recusado',
}

class Order extends PureComponent {
  constructor (props) {
    super(props)
    this.renderPayables = this.renderPayables.bind(this)
  }

  renderPayables () {
    const { payables } = this.props.order
    if (payables.length) {
      return (
        <div>
          <h4 className={style.payablesTitle}>Recebíveis </h4>
          <div className={style.payablesContainer}>
            {payables.map(payable => <Payable {...payable} key={payable.recipientId} />)}
          </div>
        </div>
      )
    }
    return null
  }

  render () {
    const { order } = this.props
    return (
      <div className={style.container}>
        <div className={style.content} >
          <h2 className={style.title}>Detalhes do pedido</h2>
          <ul className={style.transaction}>
            <li className={style.item}>
              Transação: <span className={style.value}>{order.id}</span>
            </li>
            <li className={style.item}>
              Status: <span className={style.value}>{statuses[order.status]}</span>
            </li>
            <li className={style.item}>
              Valor: <span className={style.value}>{formatCurrency(order.amount)}</span>
            </li>
          </ul>
          {this.renderPayables()}
          <div className={style.buttonContainer}>
            <ButtonLink to="/">
              Voltar a loja
            </ButtonLink>
          </div>
        </div>
      </div>
    )
  }
}

Order.defaultProps = {
  order: null,
}

Order.propTypes = {
  order: PropTypes.shape({
    id: PropTypes.number.isRequired,
    amount: PropTypes.number.isRequired,
    status: PropTypes.string.isRequired,
    payables: PropTypes.arrayOf(PropTypes.shape({
      status: PropTypes.string.isRequired,
      amount: PropTypes.number.isRequired,
      fee: PropTypes.number.isRequired,
      recipientId: PropTypes.string.isRequired,
      paymentDate: PropTypes.string.isRequired,
      dateCreated: PropTypes.string.isRequired,
    })).isRequired,
  }),
}
export default Order
