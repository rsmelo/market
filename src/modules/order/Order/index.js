import { connect } from 'react-redux'

import Order from './Order'
import {
  getOrder,
} from '../../combinedReducers'

const mapStateToProps = (state, ownProps) => ({
  order: getOrder(state, ownProps.match.params.id),
})

export default connect(mapStateToProps, {})(Order)
