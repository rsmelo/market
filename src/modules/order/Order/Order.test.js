import React from 'react'
import renderer from 'react-test-renderer'

import Order from './Order'

jest.mock('react-router-dom', () => ({ Link: 'Link' }))

describe('Order', () => {
  it('renders correctly', () => {
    const props = {
      order: {
        amount: 100,
        id: 123,
        status: 'paid',
        payables: [{
          id: '123',
          status: 'waiting_funds',
          amount: 24000,
          fee: 6000,
          recipientId: 're_cj2wd5u2600fecw6eytgcbkd0',
          paymentDate: '2017-08-01T03:00:00.000',
          dateCreated: '2017-06-29T18:16:49.411Z',
        }],
      },
      match: {
        params: {
          id: '123',
        },
      },
    }
    const tree = renderer
      .create(<Order {...props}/>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
