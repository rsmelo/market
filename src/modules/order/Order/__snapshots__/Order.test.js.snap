// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order renders correctly 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    <h2
      className={undefined}
    >
      Detalhes do pedido
    </h2>
    <ul
      className={undefined}
    >
      <li
        className={undefined}
      >
        Transação: 
        <span
          className={undefined}
        >
          123
        </span>
      </li>
      <li
        className={undefined}
      >
        Status: 
        <span
          className={undefined}
        >
          pago
        </span>
      </li>
      <li
        className={undefined}
      >
        Valor: 
        <span
          className={undefined}
        >
          R$ 100.00
        </span>
      </li>
    </ul>
    <div>
      <h4
        className={undefined}
      >
        Recebíveis 
      </h4>
      <div
        className={undefined}
      >
        <div
          className={undefined}
        >
          <ul>
            <li
              className={undefined}
            >
              ID do recebedor: 
              <span
                className={undefined}
              >
                re_cj2wd5u2600fecw6eytgcbkd0
              </span>
            </li>
            <li
              className={undefined}
            >
              Valor: 
              <span
                className={undefined}
              >
                R$ 24,000.00
              </span>
            </li>
            <li
              className={undefined}
            >
              Taxas: 
              <span
                className={undefined}
              >
                R$ 6,000.00
              </span>
            </li>
            <li
              className={undefined}
            >
              Status: 
              <span
                className={undefined}
              >
                aguardando fundos
              </span>
            </li>
            <li
              className={undefined}
            >
              Data da criação: 
              <span
                className={undefined}
              >
                29/06/2017
              </span>
            </li>
            <li
              className={undefined}
            >
              Data do pagamento: 
              <span
                className={undefined}
              >
                01/08/2017
              </span>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className={undefined}
    >
      <Link
        className=""
        to="/"
      >
        Voltar a loja
      </Link>
    </div>
  </div>
</div>
`;
